#! /bin/sh /usr/share/dpatch/dpatch-run
## 05_fix_mem_leaks_2.diff.dpatch by  <crimsun@localhost.localdomain>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix memory leaks in parser

@DPATCH@

# HG changeset patch
# User crimsun@localhost.localdomain
# Node ID 0a8e6e5e08efd89f6824ebdcec38f004508dd16a
# Parent  918f4dd50f7d88a77adb4e09d7822fa6cdf57ec8
Subject: [PATCH] [UBUNTU alsa-lib] Fix memory leaks in parser

UpstreamStatus: Applied, src/control/setup.c r1.17,
                         http://sourceforge.net/mailarchive/forum.php?thread_id=9681849&forum_id=33141

Takashi Iwai applied this patch fixing several memory leaks in the
configuration file parser.

Signed-off-by: Daniel T Chen <crimsun@fungus.sh.nu>

diff -r 918f4dd50f7d -r 0a8e6e5e08ef src/control/setup.c
--- a/src/control/setup.c	Sat Apr 29 17:41:45 2006 -0400
+++ b/src/control/setup.c	Sat Apr 29 17:49:21 2006 -0400
@@ -461,9 +461,11 @@ static int add_elem(snd_sctl_t *h, snd_c
 			err = snd_config_get_bool_ascii(tmp);
 			if (err < 0) {
 				SNDERR("field %s is not a boolean", id);
+				free(tmp);
 				goto _err;
 			}
 			lock = err;
+			free(tmp);
 			continue;
 		}
 		if (strcmp(id, "preserve") == 0) {
@@ -474,9 +476,11 @@ static int add_elem(snd_sctl_t *h, snd_c
 			err = snd_config_get_bool_ascii(tmp);
 			if (err < 0) {
 				SNDERR("field %s is not a boolean", id);
+				free(tmp);
 				goto _err;
 			}
 			preserve = err;
+			free(tmp);
 			continue;
 		}
 		if (strcmp(id, "value") == 0) {
@@ -495,9 +499,11 @@ static int add_elem(snd_sctl_t *h, snd_c
 			err = snd_config_get_bool_ascii(tmp);
 			if (err < 0) {
 				SNDERR("field %s is not a boolean", id);
+				free(tmp);
 				goto _err;
 			}
 			optional = err;
+			free(tmp);
 			continue;
 		}
 		SNDERR("Unknown field %s", id);
